fake.go

```diff
@@ -519,6 +519,57 @@ func (fake *Fake) Dump() string {
 	return buf.String()
 }
 
+// ParseDump can parse a dump for a given nft instance.
+// It expects table name and the same nftables family in all rules based on a given fake.nftContext.
+// The best way to verify that everything important was properly parsed is to
+// compare given data with nft.Dump() output.
+func (fake *Fake) ParseDump(data string) (err error) {
+	lines := strings.Split(data, "\n")
+	var i int
+	var line string
+	parsingDone := false
+	defer func() {
+		if err != nil && !parsingDone {
+			err = fmt.Errorf("line %v=`%s`: %w", i+1, line, err)
+		}
+	}()
+	tx := fake.NewTransaction()
+	for i, line = range lines {
+		line = strings.TrimSpace(line)
+		if line == "" || line[0] == '#' {
+			continue
+		}
+		words := strings.Split(line, " ")
+		if len(words) < 2 {
+			return fmt.Errorf("every command should have at least 2 words")
+		}
+		prefix := fmt.Sprintf("add %s %s %s", words[1], fake.nftContext.family, fake.nftContext.table)
+		var obj Object
+		switch words[1] {
+		case "table":
+			obj, err = parseTable(line, prefix)
```

Comment:
I think I'd like the `parse` functions as methods on the objects, for parallelism with the other per-object stuff. So you'd just do

    obj = &Table{}

here, and then after the `switch` you can do

    err = obj.parse(line, prefix)

Reply:
that is an usual signature for Parse, but I guess it makes sense here